tests/cpp/multidevice.h

```diff
@@ -48,15 +49,22 @@ class MultiDeviceTest : public NVFuserTest {
  public:
   static at::Tensor shardTensor(
       at::Tensor tensor,
-      const DeviceMesh& mesh,
+      TensorView* tv,
       DeviceIdxType deviceId) {
-    int i = 0;
-    auto devices = mesh.vector();
-    auto it = find(devices.begin(), devices.end(), deviceId);
-    if (it != devices.end()) {
-      i = std::distance(devices.begin(), it);
+    if (isSharded(tv)) {
+      auto sharded_dim = 0;
+      int i = 0;
+      auto devices = tv->getDeviceMesh().vector();
+      auto it = std::find(devices.begin(), devices.end(), deviceId);
+      if (it != devices.end()) {
+        i = std::distance(devices.begin(), it);
+      }
+      std::vector<at::indexing::TensorIndex> indices(
+          tensor.dim(), at::indexing::Slice());
+      indices[sharded_dim] = at::indexing::Slice(i, i + 1);
+      return tensor.index(indices).contiguous();
```

Comment:
For my own understanding: is it really necessary to call `contiguous()`? I guess this is done in order to make the contiguity match between this reference tensor match and the obtained sharded tensor. So this is correct if and only if the sharded tensor is also contiguous, right?

Reply:
`index` returns a view to avoid copying the tensor. `contiguous()` just ensures if the slice isn't contiguous it will create a new tensor and copy data.

Reply:
got it!